Add Room encryption docs for Kotlin SDK (#393)
powersync-ja/powersync-docs: client-sdks/orms/kotlin/room.mdx

    Room.databaseBuilder(...).setDriver(driver).build()
    ```

## Encrypted databases

> When using PowerSync with Room, PowerSync wraps the Room database instead of opening its own connection. To use an encrypted database with Room, wrap an encrypted `PersistentConnectionFactory` in a `SQLiteDriver` and pass it to `RoomDatabase.Builder.setDriver`:

Reviewer: ```suggestion
When using PowerSync with Room, PowerSync wraps the Room database instead of opening its own connection. To use an encrypted database with Room, implement `SQLiteDriver` by delegating to `PersistentConnectionFactory` and pass that to `RoomDatabase.Builder.setDriver`:
```
